program/src/processor/sweep_fees.rs

```diff
@@ -89,6 +102,46 @@ pub(crate) fn process(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramR
 
     let mut market_state = DexState::get(accounts.market)?;
     check_accounts(program_id, &market_state, &accounts)?;
+    check_metadata_account(accounts.token_metadata, &market_state.base_mint)?;
+
+    if market_state.accumulated_fees == 0 {
+        msg!("There are no fees to be extracted from the market");
+        return Err(DexError::NoOp.into());
+    }
+
+    if accounts.token_metadata.data_len() != 0 {
+        let creators = get_verified_creators(accounts.token_metadata);
+        if let Some(creators) = creators {
+            for (idx, creator) in creators.into_iter().enumerate() {
+                let token_destination = accounts.creators_token_accounts.get(idx).unwrap();
+                let amount = market_state.accumulated_royalties * (creator.share as u64) / 100;
```

Comment:
If token metadata creator share data is wrong, this could lead to a larger token transfer than the total amount of fees, mayber we should check that the sum is 100?